### PR TITLE
Add support for macOS

### PR DESCRIPTION
**Pull request by Vertexwahn**
This PR adds support for macOS

### CODE REVIEW

**Comment by Vertexwahn** [comment]
With a few hacks `bazel build --cxxopt=-std=c++17 //...` works now on macOS.

If I try to run `bazel run --cxxopt=-std=c++17 //tests/qt_resource:main` I get the following error:

```
dyld: Symbol not found: __ZN10QByteArray6_emptyE
```

This is a runtime error

**Comment by Vertexwahn** [comment]
@justbuchanan Done. Please merge.

File: qt.BUILD [hunk 1]
```diff
@@ -50,13 +50,29 @@ QT_LIBRARIES = [
     for name, include_folder, _, _ in QT_LIBRARIES
 ]
 
+[
+    cc_library(
+        name = "qt_%s_osx" % name,
+        # When being on Windows or Linux this glob will be empty
```
Comment (justbuchanan) [inline qt.BUILD:56]: Can we omit the `hdrs` parameter here since it will be empty anyways?

**Comment by justbuchanan** [comment]
Nice work, thanks for adding this!

**Pull request merged**